GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -99,7 +99,7 @@ public void Dispose()
 
             request.Headers.UserAgent.Add(this.userAgentHeader);
 
-            if (!string.IsNullOrEmpty(authString))
+            if (!this.authentication.IsAnonymous)

```

Comment:
This is definitely cleaner than the old code, but for my own understanding is `authString` always null or empty when `authentication.IsAnonymous` is true?

Reply:
> is authString always null or empty 

Yes - a couple of lines above we only attempt to get credentials if this is *not* anonymous access. It is a bit removed and I asked myself the same thing. 

I had wondered if we could do a bit broader cleanup / refactoring to improve readability around some of the authentication classes to help address questions like this, but that is a different topic :)